plugins/org.eclipse.elk.alg.packing.rectangles/src/org/eclipse/elk/alg/packing/rectangles/firstiteration/BestOptionFinder.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Kiel University and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Kiel University - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.elk.alg.packing.rectangles.firstiteration;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.elk.alg.packing.rectangles.util.DrawingData;
+
+/**
+ * Compares placement options, i.e. different drawings and offers methods to find the best options out of the given
+ * ones.
+ * 
+ * @author dalu
+ */
+public final class BestOptionFinder {
+
+    private BestOptionFinder() {
+    }
+
+    /**
+     * Filters the given list for the element with the aspect ratio that is closest to the given desired aspect ratio.
+     * Returns a list with the elements that have smallest deviation to the desired aspect ratio as multiple options may
+     * have the same aspect ratio.
+     * 
+     * @param candidates
+     *            list of still considerable options.
+     * @param dar
+     *            desired aspect ratio.
+     * @return list of elements with aspect ratio being as close to the desired aspect ratio as possible.
+     */
+    protected static List<DrawingData> findBestAspectRatio(final List<DrawingData> candidates, final double dar) {
+        List<DrawingData> remainingCandidates = new ArrayList<DrawingData>();
+        double smallestDeviation = Double.MAX_VALUE;
```

Comment:
Better to use `Double.POSITIVE_INFINITY` (since `Double.MIN_VALUE` behaves different from what one would expect knowing `Integer.MIN_VALUE`. The first one is the smalles non-negative.)